demo/app/public/index.php

```diff
@@ -32,8 +32,10 @@
     Request::setTrustedHosts(explode(',', $trustedHosts));
 }
 
-$kernel = new Kernel($env, $debug);
-$request = Request::createFromGlobals();
-$response = $kernel->handle($request);
-$response->send();
-$kernel->terminate($request, $response);
+Amp\Loop::run(\Amp\GreenThread\coroutine(function () use($env, $debug) {
+    $kernel = new Kernel($env, $debug);
```

Comment:
Why not create the kernel _outside_ the event loop to avoid restarting it on every request?

Reply:
This code is run under fpm sapi, so it will be restarted for every request no matter inside or outside the event loop.

But this would be a possible Step 4, to remove nginx + fpm and use the amp http server instead, and then the kernel could be created before yes.

Also in this context, it allows the kernel to do async calls (maybe you want to load configuration in parallel ?)

Reply:
My quest is more about a native handling of both HTTP & WS requests with the same PHP application than loading files asynchronously 🤣

Reply:
Unfortunetaly this is not the goal of this demo, you should use amp http server for this